### PR TITLE
selfhost+formatter: format everything, take 2?

### PR DESCRIPTION
**Pull request by alimpfard**
Let's see what this looks like :P

### CODE REVIEW

**Comment by AtkinsSJ** [comment]
Looks like it's inserting two spaces between `unsafe` and the `{`.

File: selfhost/cpp_import/clang.jakt [hunk 1]
```diff
@@ -40,7 +38,7 @@ import cpp_import::clang_c {
     clang_parseTranslationUnit2, clang_saveTranslationUnit, clang_tokenize, clang_visitChildren
 }
 import utility { allocate }
-import extern c "stdlib.h" {
+import extern c"stdlib.h" {
```
Comment (alimpfard) [inline selfhost/cpp_import/clang.jakt:41]: Missing space after `c`.

File: selfhost/cpp_import/clang.jakt [hunk 2]
```diff
@@ -49,7 +47,7 @@ import jakt::prelude::configuration { UserConfiguration }
 fn bitcast<D, S>(anon value: S) -> D {
     mut x = null<D>()
     let f = fn[&x, &value]() {
-        unsafe { cpp { "x = bit_cast<RemoveReference<decltype(x)>>(value);" } }
+        unsafe  { cpp { "x = bit_cast<RemoveReference<decltype(x)>>(value);" } }
```
Comment (alimpfard) [inline selfhost/cpp_import/clang.jakt:50]: Extra space in `unsafe {`.

File: selfhost/cpp_import/clang.jakt [hunk 3]
```diff
@@ -216,7 +216,7 @@ struct TemplateParameters {
     packs_by_name: [String]
 }
 
-struct HashableCursor implements(Equal<HashableCursor>, Hashable) {
+struct HashableCursor implements(Equal<HashableCursor>,Hashable) {
```
Comment (alimpfard) [inline selfhost/cpp_import/clang.jakt:219]: Missing space in `implements(..., ...)`

File: selfhost/cpp_import/clang.jakt [hunk 4]
```diff
@@ -441,7 +447,9 @@ class CppImportProcessor {
                         let base_usr = string_from(clang_getCursorUSR(cursor: definition_cursor))
 
                         // NonnullRefPtr<T> where T is a class -> T
-                        let is_nnrp_class_type = base_usr.starts_with("c:@N@AK@S@NonnullRefPtr>") or base_usr == "c:@N@AK@ST>1#T@NonnullRefPtr"
+                        let is_nnrp_class_type = base_usr.starts_with(
+                            "c:@N@AK@S@NonnullRefPtr>"
+                        ) or base_usr == "c:@N@AK@ST>1#T@NonnullRefPtr"
```
Comment (alimpfard) [inline selfhost/cpp_import/clang.jakt:452]: :thinking: maybe break from the middle?

File: selfhost/cpp_import/clang.jakt [hunk 5]
```diff
@@ -842,7 +850,7 @@ class CppImportProcessor {
         let name = name_from(cursor)
         let template_parameters = .extract_cursors_of_kinds(
             cursor
-            kinds: {CXCursorKind::CXCursor_TemplateTypeParameter, CXCursorKind::CXCursor_NonTypeTemplateParameter}
+            kinds:  {CXCursorKind::CXCursor_TemplateTypeParameter, CXCursorKind::CXCursor_NonTypeTemplateParameter }
```
Comment (alimpfard) [inline selfhost/cpp_import/clang.jakt:853]: Extra space before `{` _and_ `}`.

File: selfhost/cpp_import/clang.jakt [hunk 6]
```diff
@@ -1779,8 +1791,8 @@ class CppImportProcessor {
         }
 
         for base in .extract_cursors_of_kinds(
-            cursor: definition
-            kinds: {CXCursorKind::CXCursor_CXXBaseSpecifier}
+                cursor: definition
+                kinds:  {CXCursorKind::CXCursor_CXXBaseSpecifier }
```
Comment (alimpfard) [inline selfhost/cpp_import/clang.jakt:1795]: Indented too much

File: selfhost/cpp_import/clang_c.jakt [hunk 7]
```diff
@@ -1,9 +1,9 @@
-import extern c "clang-c/Index.h" {
-    extern struct CXIndex { fn CXIndex() -> CXIndex }
-    extern struct CXTranslationUnit { fn CXTranslationUnit() -> CXTranslationUnit }
-    extern struct CXCursor { fn CXCursor() -> CXCursor }
+import extern c"clang-c/Index.h" {
+    extern struct CXIndex { fn CXIndex() -> CXIndex}
+    extern struct CXTranslationUnit { fn CXTranslationUnit() -> CXTranslationUnit}
+    extern struct CXCursor { fn CXCursor() -> CXCursor}
```
Comment (alimpfard) [inline selfhost/cpp_import/clang_c.jakt:4]: Missing space before `}`.

File: selfhost/cpp_import/libcpp.jakt [hunk 8]
```diff
@@ -122,7 +120,7 @@ import extern "LibCpp/Parser.h" {
 }
 
 trait DeclarationImpl {}
-type Cpp::Declaration implements(DeclarationImpl) {
+type Cpp:: Declarationimplements ( DeclarationImpl)  {
```
Comment (alimpfard) [inline selfhost/cpp_import/libcpp.jakt:123]: Extra spaces everywhere in `type ... implements(...)` decl.

File: selfhost/cpp_import/vector.jakt [hunk 9]
```diff
@@ -13,7 +13,7 @@ import extern "AK/Vector.h" {
 
 use AK::Vector
 
-struct VectorIterator<T> implements(Iterable<T>) {
+struct VectorIterator<T>implements ( Iterable< T> )  {
```
Comment (alimpfard) [inline selfhost/cpp_import/vector.jakt:16]: missing space before `implements`.

File: selfhost/ide.jakt [hunk 10]
```diff
@@ -117,19 +119,46 @@ fn record_to_symbol(anon record: ParsedRecord) throws -> JaktSymbol {
     let record_kind = match record.record_type {
         Struct(fields) => {
             for field in fields {
-                children.push(JaktSymbol(name: field.var_decl.name, detail: None, kind: "field", range: field.var_decl.span, selection_range: field.var_decl.span, children: []))
+                children.push(
+                    JaktSymbol(
+                    name: field.var_decl.name
```
Comment (alimpfard) [inline selfhost/ide.jakt:124]: Not indented enough.

File: selfhost/ide.jakt [hunk 11]
```diff
@@ -738,9 +819,9 @@ fn find_span_in_expression(program: CheckedProgram, expr: CheckedExpression, spa
                         if marker_span.contains(span) {
                             // FIXME: return Some(get_enum_variant_usage_from_type_id_and_name(program, type_id: subject_type_id, variant_index))
                             return Some(get_enum_variant_usage_from_type_id_and_name(
-                                    program
-                                    type_id: subject_type_id
-                                    name))
+                                program
+                                type_id: subject_type_id
+                            name))
```
Comment (alimpfard) [inline selfhost/ide.jakt:824]: - Not indented enough
- Maybe move the `))` to the next line?

**Review by alimpfard — COMMENTED** [review]
Up to interpreter.jakt

**Pull request closed without merging**